core/corehttp/gateway_test.go

```diff
@@ -114,21 +130,28 @@ func TestGatewayGet(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	ns["/ipns/example.com"] = path.FromString("/ipfs/" + k)
+	kTTL := 10 * time.Minute
+	ns["/ipns/example.com"] = mockEntry{
+		path: path.FromString("/ipfs/" + k),
```

Comment:
completely out of scope of this PR, but I'm thinking we should probably make `FromString` do the parsing check in `path.ParsePath`, and panic if it fails.